lib/phobos.rb

```diff
@@ -31,19 +31,46 @@
 
 Thread.abort_on_exception = true
 
+Logging.init :debug, :info, :warn, :error, :fatal
+
+# Monkey patch to fix this issue: https://github.com/zendesk/ruby-kafka/pull/732
+module Logging
+  # :nodoc:
+  class Logger
+    # :nodoc:
+    def formatter=(*args); end
+
+    # :nodoc:
+    def push_tags(*args); end
+
+    # :nodoc:
+    def pop_tags(*args); end
+  end
+end
+
 module Phobos
   class << self
     attr_reader :config, :logger
     attr_accessor :silence_log
 
     def configure(configuration)
       @config = fetch_configuration(configuration)
-      @config.class.send(:define_method, :producer_hash) { Phobos.config.producer&.to_hash }
-      @config.class.send(:define_method, :consumer_hash) { Phobos.config.consumer&.to_hash }
+      config_define_methods
       @config.listeners ||= []
       configure_logger
     end
 
+    def config_define_methods # rubocop:disable Metrics/AbcSize
+      @config.class.send(:define_method, :cache_sync_producer) do
+        hash = Phobos.config.producer&.to_hash
+        hash[:cache_sync_producer]
+      end
+      @config.class.send(:define_method, :producer_hash) do
+        Phobos.config.producer&.to_hash&.except(:cache_sync_producer)
```

Comment:
I don't think we need to exclude this option here, it just increases ABC size, and it is confusing that what we set in config yml doesnt show up in Phobos.config

Reply:
Yeah, I was thinking of this as "this is a producer configuration so it belongs under producer", but I realize that it does mean we have to tease them apart internally. What do you think about adding a `phobos_producer` config heading which would just contain this setting for now?

Reply:
Agreed, see other comment I made where I suggested name `INTERNAL_PRODUCER_PARAMS`